tests/test_helpers/test_dialogue/test_base.py

```diff
@@ -280,102 +257,331 @@ def test_dialogue_properties(self):
         assert self.dialogue.last_incoming_message is None
         assert self.dialogue.last_outgoing_message is None
         assert self.dialogue.last_message is None
-        assert self.dialogue.get_message(3) is None
+
         assert self.dialogue.is_empty
 
-    def test_update_positive(self):
-        """Positive test for the 'update' method."""
-        valid_initial_msg = DefaultMessage(
+    def test_counterparty_from_message(self):
+        """Test the 'counterparty_from_message' method."""
+        message_1 = DefaultMessage(
             dialogue_reference=(str(1), ""),
             message_id=1,
             target=0,
             performative=DefaultMessage.Performative.BYTES,
             content=b"Hello",
         )
-        valid_initial_msg.counterparty = self.opponent_address
+        message_1.sender = self.agent_address
+        message_1.to = self.opponent_address
 
-        assert self.dialogue.update(valid_initial_msg)
-        assert self.dialogue.last_outgoing_message == valid_initial_msg
-
-    def test_update_sets_missing_counterparty(self):
-        """Test the 'update' method sets the missing counterparty field of the input message."""
-        valid_initial_msg = DefaultMessage(
+        message_2 = DefaultMessage(
             dialogue_reference=(str(1), ""),
             message_id=1,
             target=0,
             performative=DefaultMessage.Performative.BYTES,
             content=b"Hello",
         )
+        message_2.sender = self.opponent_address
+        message_2.to = self.agent_address
 
-        assert self.dialogue.update(valid_initial_msg)
-        assert self.dialogue.last_outgoing_message == valid_initial_msg
         assert (
-            self.dialogue.last_outgoing_message.counterparty
-            == self.dialogue.dialogue_label.dialogue_opponent_addr
+            self.dialogue.counterparty_from_message(message_1) == self.opponent_address
+        )
+        assert (
+            self.dialogue.counterparty_from_message(message_2) == self.opponent_address
         )
 
-    def test_update_negative_not_is_extendible(self):
-        """Negative test for the 'update' method: dialogue is not extendable with the input message."""
-        invalid_message_id = DefaultMessage(
+    def test_is_message_by_self(self):
+        """Test the 'is_message_by_self' method."""
+        message_1 = DefaultMessage(
             dialogue_reference=(str(1), ""),
-            message_id=0,
+            message_id=1,
             target=0,
```

Comment:
Throughout you can drop message id and target when they are equal to the default values

Reply:
Done